content/graph/LCT.h

```diff
@@ -0,0 +1,116 @@
+/**
+ * Author: 
+ * Description: link-cut Tree. Supports BBST-like 
+ * augmentations. (Can be used in place of HLD).
+ * Current implementation supports update value at a node,
+ * and query max on a path.
+ * Tested on: http://acm.timus.ru/problem.aspx?num=1553
+ * Status: Passes existing fuzz tests (with function names modified).
+ */
+struct Node {
+	bool flip = 0;
+	Node *pp, *p, *c[2];
```

Comment:
is it possible to add a one-line comment explaining what p and pp are? or are they standard names?

Reply:
p = parent, pp = path-parent (and c = children).
p and c are straight forward. path-parent is from the Wikipedia article on LCTs and also this set of lecture notes https://courses.csail.mit.edu/6.851/spring12/scribe/L19.pdf (which I think the Wikipedia page is based on). Besides writing out the acronyms, do you think it's a good idea to explain what a path-parent is? I think that if you understand how LCTs work (by decomposing a tree into preferred paths), then knowing that pp stands for path-pointer should be enough information.

Reply:
So something like `// p ~ parent in segtree, pp = path parent`? (Now, I'm also not sure whether that's helpful information when augmenting... but I could imagine such a comment being helpful for understanding)